mathematical-companion/main.tex

```diff
@@ -0,0 +1,1300 @@
+
+In fact, in a complete Checksum Worksheet, \emph{every single cell} that the user
+fills in is an affine function of the share data. This means that if you pick an
+arbitrary cell, say, the fourth cell of the tenth row, you can use the above process
+to combine those cells from the initial shares' worksheets and produce the
+corresponding cell on the derived share's worksheet.
+
+This means in particular, that the final row (\vc{SECRETSHARE32}) will be
+preserved, meaning that the checksum of the derived share will be correct. But it
+also means that, if the user uses the checksum worksheet to verify her share
+translation (which she should!) she can ``sanity check'' her work by pre-deriving
+cells.
+
+This is was an exciting realization, because normally the Checksum Worksheet takes
+a long time to fill out and provides the user no feedback until the very end, at
+which point the result might just be ``wrong residue, start over''. To avoid this
+frustration, we recommend the following process for derived shares and recovery:
+\begin{enumerate}
+\item Before deriving any shares, complete Checksum Worksheets for the inital
+shares. (During construction these should be readily available; during recovery
+it's worth doing as a sanity check.)
+\item Derive the actual share, using the above method.
+\item Copy the share into the top diagonal, i.e.~the bolded data cells, of a fresh
+Checksum Worksheet.
+\item Derive the cells of the \emph{bottom} diagonal directly, in the same way that
+you derived the top diagonal.
+\item Start working through the worksheet.
+\end{enumerate}
+
+If, at any point, the computed value of a bottom diagonal square doesn't match
+the precomputed value, it means you made a mistake in that column. Re-derive the
+top and bottom values and redo the additions before continuing.
+
+\section{Quickchecks}
+
+\paragraph{Important Note.} {\color{BrickRed}This section describes a series of
+worksheets which, as of March 2023, do not actually exist.}
+
+Recall that the generating polynomial for our primary code is
+\begin{align*}
+    G(x) = x^{13}
+      &+ \vc{E}x^{12} + \vc{M}x^{11} + \vc{3}x^{10} + \vc{G}x^9 + \vc{Q}x^8 + \vc{E}x^7 \\
+      &+ \vc{E}x^6 + \vc{E}x^5 + \vc{L}x^4 + \vc{M}x^3 + \vc{C}x^2 + \vc{S}x + \vc{S}.
+\end{align*}
+Previously we simply took this polynomial as a given. But in this section and the
+next, we need to dig a bit deeper into how it was constructed.
+
+To this end we first construct the extension field $\ftttwo$. Just as we
+produced $\fttwo$ by adding a root of the irreducible 5th-degree polynomial
+$x^5 + x^3 + 1$ over $\ftwo$, we can adjoin a root of the irreducible
+2nd-degree polynomial $x^2 + x + 1$ to $\fttwo$ to get a new field $\ftttwo$.
+We will call this new root $\zeta$. By construction, $\zeta$ satisfies
+the equation $\zeta^2 = \zeta + 1$.
+
+We can write any element of $\ftttwo$ as $a\zeta + b$, where $a$ and $b$ are
```

Comment:
I think `a + b\zeta` would be more traditional.  I guess I could be wrong.

Reply:
Hmm, yeah, that is how we write complex numbers isn't it. I'll go over the doc and reverse the ordering here.